legate/driver/args.py

```diff
@@ -42,6 +46,45 @@
 __all__ = ("parser",)
 
 
+def detect_multi_node_defaults() -> tuple[dict[str, Any], dict[str, Any]]:
+    from os import getenv
+
+    nodes_kw = dict(NODES.kwargs)
+    ranks_per_node_kw = dict(RANKS_PER_NODE.kwargs)
+    where = None
+
+    if nodes_env := getenv("OMPI_COMM_WORLD_SIZE"):
+        if ranks_per_node_env := getenv("OMPI_COMM_WORLD_LOCAL_SIZE"):
+            nodes, ranks_per_node = int(nodes_env), int(ranks_per_node_env)
+            where = "OMPI"
+
+    elif nodes_env := getenv("MV2_COMM_WORLD_SIZE"):
+        if ranks_per_node_env := getenv("MV2_COMM_WORLD_LOCAL_SIZE"):
+            nodes, ranks_per_node = int(nodes_env), int(ranks_per_node_env)
+            where = "MV2"
+
+    elif nodes_env := getenv("SLURM_JOB_NUM_NODES"):
+        if ranks_env := getenv("SLURM_NTASKS"):
+            nodes, ranks = int(nodes_env), int(ranks_env)
+            assert ranks % nodes == 0
+            ranks_per_node = ranks // nodes
+            where = "SLURM"
+
+    else:
+        nodes = defaults.LEGATE_NODES
+        ranks_per_node = defaults.LEGATE_RANKS_PER_NODE
+
+    nodes_kw["default"] = nodes
+    ranks_per_node_kw["default"] = ranks_per_node
```

Comment:
Since the defaults are already specified in `util.shared_args.NODES` and `util.shared_args.RANKS_PER_NODE`, you could just leave them alone if `where` is None: 
```suggestion
    if where:
        nodes_kw["default"] = nodes
        ranks_per_node_kw["default"] = ranks_per_node
```

Reply:
This occurred to me but I would prefer to keep an explicit catch-all "else" clause

Reply:
https://github.com/nv-legate/legate.core/pull/832/commits/2975cc7ab7304e8f068a2707096ced9c12e4486e